snap/snapcraft.yaml

```diff
@@ -0,0 +1,63 @@
+name: tempest
+adopt-info: tempest
+summary: OpenStack Integration Test Suite
+description: |
+ Tempest is a set of integration tests to be run against a live
+ OpenStack cluster. Tempest has batteries of tests for OpenStack
+ API validation, Scenarios, and other specific tests useful in
+ validating an OpenStack deployment.
+confinement: strict
+grade: stable
+base: core22
+
+apps:
+  tempest:
+    command: bin/tempest
+    completer: usr/share/bash-completion/completions/tempest
+    plugs:
+      - network-observe
+      - network
+      - home
+parts:
+  tempest:
+    plugin: python
+    source: .
+    python-packages:
+      - tempest
+      - barbican-tempest-plugin
+      - blazar-tempest-plugin
+      - cinder-tempest-plugin
+      - cloudkitty-tempest-plugin
+      - cyborg-tempest-plugin
+      - designate-tempest-plugin
+      - ec2api-tempest-plugin
+      - freezer-tempest-plugin
+      - glance-tempest-plugin
+      - heat-tempest-plugin
+      - ironic-tempest-plugin
+      - keystone-tempest-plugin
+      - kuryr-tempest-plugin
+      - magnum-tempest-plugin
+      - manila-tempest-plugin
+      - mistral-tempest-tests
+      - monasca-tempest-plugin
+      - murano-tempest-plugin
+      - neutron-tempest-plugin
+      - octavia-tempest-plugin
+      - oswin-tempest-plugin
+      - sahara-tests
+      - senlin-tempest-plugin
+      - solum-tempest-plugin
+      - telemetry-tempest-plugin
+      - trove-tempest-plugin
+      - venus-tempest-plugin
+      - vitrage-tempest-plugin
+      - watcher-tempest-plugin
+      - zaqar-tempest-plugin
+      - zun-tempest-plugin
+    override-prime: |
+      craftctl default
+      craftctl set version="$(tempest --version | awk '{print $2}')"
+      mkdir -p usr/share/bash-completion/completions
+      bin/tempest complete > usr/share/bash-completion/completions/tempest
+      echo "complete -F _tempest tempest.tempest" >> usr/share/bash-completion/completions/tempest
```

Comment:
Not sure why following comment missed last review, maybe I cancel it.

Do you know why this is here? I do not see this in the [overview](https://docs.openstack.org/tempest/latest/overview.html) and I'm curious if it's really needed.

Reply:
Yes, I put it there to make bash completion work for the tempest binary exposed by the snap. The cliff package provides a mechanism to generate a bash-completion script by default, so `tempest complete` produces that script. Another example of this usage can be seen in the snap-openstackclients [source code](https://opendev.org/x/snap-openstackclients/src/commit/4c0fdd1d73ef3f2dcbd66950f34452178c424578/snapcraft.yaml#L200)

Reply:
Oh I see, thanks for explanation :+1: .